diff -ruN '--exclude=.git' ns-3-dev/CMakeLists.txt ns-3-patched/CMakeLists.txt
--- ns-3-dev/CMakeLists.txt	2023-05-02 16:01:44.221536607 +0200
+++ ns-3-patched/CMakeLists.txt	2023-05-02 15:58:54.783816242 +0200
@@ -79,6 +79,7 @@
 )
 option(NS3_PYTHON_BINDINGS "Build ns-3 python bindings" OFF)
 option(NS3_SQLITE "Build with SQLite support" ON)
+option(NS3_CURL "Build with cURL support" ON)
 option(NS3_EIGEN "Build with Eigen support" ON)
 option(NS3_STATIC "Build a static ns-3 library and link it against executables"
        OFF
diff -ruN '--exclude=.git' ns-3-dev/build-support/custom-modules/ns3-configtable.cmake ns-3-patched/build-support/custom-modules/ns3-configtable.cmake
--- ns-3-dev/build-support/custom-modules/ns3-configtable.cmake	2023-05-02 16:01:44.221536607 +0200
+++ ns-3-patched/build-support/custom-modules/ns3-configtable.cmake	2023-05-02 16:00:22.268971508 +0200
@@ -184,6 +184,9 @@
   string(APPEND out "SQLite support                : ")
   check_on_or_off("NS3_SQLITE" "ENABLE_SQLITE")
 
+  string(APPEND out "cURL support                  : ")
+  check_on_or_off("NS3_CURL" "ENABLE_CURL")
+  
   string(APPEND out "Eigen3 support                : ")
   check_on_or_off("NS3_EIGEN" "ENABLE_EIGEN")
 
diff -ruN '--exclude=.git' ns-3-dev/build-support/macros-and-definitions.cmake ns-3-patched/build-support/macros-and-definitions.cmake
--- ns-3-dev/build-support/macros-and-definitions.cmake	2023-05-02 16:02:36.132855438 +0200
+++ ns-3-patched/build-support/macros-and-definitions.cmake	2023-05-02 15:59:39.396445183 +0200
@@ -797,6 +797,24 @@
     endif()
   endif()
 
+  set(ENABLE_CURL False)
+  if(${NS3_CURL})
+    # find_package(curl QUIET) # unsupported in CMake 3.10 We emulate the
+    # behavior of find_package below
+    find_external_library(
+      DEPENDENCY_NAME curl
+      HEADER_NAME curl.h
+      LIBRARY_NAME curl
+    )
+
+    if(${curl_FOUND})
+      set(ENABLE_CURL True)
+      include_directories(${curl_INCLUDE_DIRS})
+    else()
+      message(${HIGHLIGHTED_STATUS} "cURL was not found")
+    endif()
+  endif()
+
   set(ENABLE_EIGEN False)
   if(${NS3_EIGEN})
     find_package(Eigen3 QUIET)
